crates/typst/src/foundations/args.rs

```diff
@@ -279,6 +291,29 @@ impl Args {
         args.take()
     }
 
+    /// Returns the positional argument at the specified index, or the named
+    /// argument with the specified name.
+    ///
+    /// If the key is an [integer]($int), this is equivalent to first calling
+    /// [`pos`]($arguments.pos) and then [`array.at`]. If it is a [string]($str),
+    /// this is equivalent to first calling [`named`]($arguments.named) and then
+    /// [`dictionary.at`].
+    #[func]
+    pub fn at(
+        &self,
+        /// The index or name of the argument to get.
+        key: ArgumentKey,
+        /// A default value to return if the key is invalid.
+        #[named]
+        default: Option<Value>,
+    ) -> StrResult<Value> {
+        match &key {
+            ArgumentKey::Index(index) => self.to_pos().at(*index, default).ok(),
+            ArgumentKey::Name(name) => self.to_named().at(name.clone(), default).ok(),
```

Comment:
I think if we already have this method, we should make it more efficient than calling `to_pos` and `to_named` internally, since those are quite costly.

Reply:
Done!